pkg/metrics/metrics.go

```diff
@@ -0,0 +1,36 @@
+package metrics
+
+import (
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+// ReconcileAttempts is a counter that indicates the number of attempts to reconcile the operator configuration.
+var ReconcileAttempts = metrics.NewCounter(&metrics.CounterOpts{
+	Name:           "cluster_monitoring_operator_reconcile_attempts_total",
+	Help:           "Number of attempts to reconcile the operator configuration",
+	StabilityLevel: metrics.ALPHA,
+})
+
+// ReconcileStatus is a gauge that indicates the latest reconciliation state.
+var ReconcileStatus = metrics.NewGauge(&metrics.GaugeOpts{
+	Name:           "cluster_monitoring_operator_last_reconciliation_successful",
+	Help:           "Latest reconciliation state. Set to 1 if last reconciliation succeeded, else 0.",
+	StabilityLevel: metrics.ALPHA,
+})
+
+// ConfiguredCollectionProfile is a gauge that indicates the configured collection profile.
+var ConfiguredCollectionProfile = metrics.NewGaugeVec(&metrics.GaugeOpts{
+	Name:           "cluster_monitoring_operator_configured_collection_profile",
+	Help:           "The configured collection profile",
+	StabilityLevel: metrics.ALPHA,
+	// TODO: Use LabelValueAllowLists, once GA'd.
```

Comment:
Can you expand a bit? I'm not sure to know to which component "once GA'd" applies :)

Reply:
ACK, updated.